flutter/shell/platform/tizen/tizen_renderer_vulkan.cc

```diff
@@ -0,0 +1,972 @@
+  } else {
+    return VK_ERROR_EXTENSION_NOT_PRESENT;
+  }
+}
+
+void DestroyDebugUtilsMessengerEXT(VkInstance instance,
+                                   VkDebugUtilsMessengerEXT debugMessenger,
+                                   const VkAllocationCallbacks* pAllocator) {
+  auto func = (PFN_vkDestroyDebugUtilsMessengerEXT)vkGetInstanceProcAddr(
+      instance, "vkDestroyDebugUtilsMessengerEXT");
+  if (func != nullptr) {
+    func(instance, debugMessenger, pAllocator);
+  }
+}
+
+TizenRendererVulkan::TizenRendererVulkan(TizenViewBase* view) {
+  InitVulkan(view);
+}
+
+bool TizenRendererVulkan::InitVulkan(TizenViewBase* view) {
+  if (!CreateInstance()) {
+    FT_LOG(Error) << "Failed to create Vulkan instance";
+    return false;
+  }
+  if (enable_validation_layers_) {
+    SetupDebugMessenger();
+  }
+
+  if (!TizenRenderer::CreateSurface(view)) {
+    FT_LOG(Error) << "Failed to create surface";
+    return false;
+  }
+  if (!PickPhysicalDevice()) {
+    FT_LOG(Error) << "Failed to pick physical device";
+    return false;
+  }
+  if (!CreateLogicalDevice()) {
+    FT_LOG(Error) << "Failed to create logical device";
+    return false;
+  }
+  if (!GetDeviceQueue()) {
+    FT_LOG(Error) << "Failed to get device queue";
+    return false;
+  }
+  if (!CreateSemaphore()) {
+    FT_LOG(Error) << "Failed to create semaphore";
+    return false;
+  }
+  if (!CreateFence()) {
+    FT_LOG(Error) << "Failed to create fence";
+    return false;
+  }
+  if (!CreateCommandPool()) {
+    FT_LOG(Error) << "Failed to create command pool";
+    return false;
+  }
+  if (!InitializeSwapchain()) {
+    FT_LOG(Error) << "Failed to initialize swapchain";
+    return false;
+  }
+  is_valid_ = true;
+  return true;
+}
+
+void TizenRendererVulkan::Cleanup() {
+  if (logical_device_) {
+    // Ensure all GPU work is complete before destroying anything.
+    vkDeviceWaitIdle(logical_device_);
+
+    for (size_t i = 0; i < present_transition_buffers_.size(); ++i) {
+      vkFreeCommandBuffers(logical_device_, swapchain_command_pool_, 1,
+                           &present_transition_buffers_[i]);
+    }
+    if (swapchain_ != VK_NULL_HANDLE) {
+      vkDestroySwapchainKHR(logical_device_, swapchain_, nullptr);
+      swapchain_ = VK_NULL_HANDLE;
+    }
+    DestroyCommandPool();
+    vkDestroyFence(logical_device_, image_ready_fence_, nullptr);
+    vkDestroySemaphore(logical_device_, present_transition_semaphore_, nullptr);
+    vkDestroyDevice(logical_device_, nullptr);
+    logical_device_ = VK_NULL_HANDLE;
+  }
+  DestroySurface();
+  if (enable_validation_layers_) {
+    DestroyDebugUtilsMessengerEXT(instance_, debug_messenger_, nullptr);
+  }
+  if (instance_ != VK_NULL_HANDLE) {
+    vkDestroyInstance(instance_, nullptr);
+    instance_ = VK_NULL_HANDLE;
+  }
+}
+
+std::unique_ptr<ExternalTexture> TizenRendererVulkan::CreateExternalTexture(
+    const FlutterDesktopTextureInfo* texture_info) {
+  return nullptr;
```

Comment:
Question
External texture is not supported?

Reply:
I have finished code of External texture:
1. Engine PR : https://github.com/flutter-tizen/flutter/pull/17
2. Embedder code : https://github.com/xiaowei-guan/embedder/commit/6a94746b5a006730e6f5280d5a4fecf8eef53b4d
After vulkan backend PR released, I will create a new PR for texture.